elements/radio-group/radio-group.js

```diff
@@ -1,151 +1,241 @@
+     * the one passed as an argument. If no next `RadioButton` is found, return
+     * null.
+     */
+    _nextRadioButton(node) {
+      let next = node.nextElementSibling;
+      while (next) {
+        if (next.getAttribute('role') === 'radio') {
+          return next;
+        }
+        next = next.nextElementSibling;
       }
-      next = next.nextElementSibling;
+      return null;
     }
 
-    return null;
-  }
-
-  _setCheckedToPrevButton() {
-    if (this.selectedRadioButton === this.firstRadioButton) {
-      this._setChecked(this.lastRadioButton);
-    } else {
-      this._setChecked(this._prevRadioButton(this.selectedRadioButton));
+    /**
+     * This method is called in response to a user pressing a key to move
+     * backwards through the `RadioGroup`.
+     * Check to see if the currently checked `RadioButton` is the first child.
+     * If so, loop around and focus the last child. Otherwise, find the previous
+     * sibling of the currently checked `RadioButton`, and make it the new
+     * checked button.
+     */
+    _setCheckedToPrevButton() {
+      let checkedButton = this.checkedRadioButton;
+      if (checkedButton === this.firstRadioButton) {
+        this._setChecked(this.lastRadioButton);
+      } else {
+        this._setChecked(this._prevRadioButton(checkedButton));
+      }
     }
-  }
 
-  _setCheckedToNextButton() {
-    if (this.selectedRadioButton === this.lastRadioButton) {
-      this._setChecked(this.firstRadioButton);
-    } else {
-      this._setChecked(this._nextRadioButton(this.selectedRadioButton));
+    /**
+     * This method is called in response to a user pressing a key to move
+     * forwards through the `RadioGroup`.
+     * Check to see if the currently checked `RadioButton` is the last child.
+     * If so, loop around and focus the first child. Otherwise, find the next
+     * sibling of the currently checked `RadioButton`, and make it the new
+     * checked button.
+     */
+    _setCheckedToNextButton() {
+      let checkedButton = this.checkedRadioButton;
+      if (checkedButton === this.lastRadioButton) {
+        this._setChecked(this.firstRadioButton);
+      } else {
+        this._setChecked(this._nextRadioButton(checkedButton));
+      }
     }
-  }
 
-  setChecked(node) {
-    // Unset _all_ radio buttons that are currently in the group.
-    const radioButtons = Array.from(this.querySelectorAll('[role="radio"]'));
-    for (let btn of radioButtons) {
-      let btn = radioButtons[i];
-      btn.setAttribute('aria-checked', 'false');
-      btn.tabIndex = -1;
+    /**
+     * Any user action (a keypress or mouse click) eventually funnels down to
+     * this method which ensures that only the passed in element is checked.
+     * Uncheck _all_ `RadioButton` children. Then set the `RadioButton` that was
+     * passed in to `aria-checked=true`. Also make it focusable with
+     * `tabIndex=0` and call its `focus()` method.
+     */
+    _setChecked(node) {
```

Comment:
Sorry, didn’t see this earlier – but maybe make this method public? Or is there another JS API to set the checked radio that I am missing?